onnxruntime/core/graph/graph_proto_serializer.cc

```diff
@@ -43,12 +43,33 @@
   }
 
   if (include_initializer) {
+    std::unordered_set<std::string> current_scope_initializer_set;
+
     auto& initializers = graph_view.GetAllInitializedTensors();
     for (auto& it : initializers) {
       auto* p_initializer = graph_proto.add_initializer();
       *p_initializer = *(it.second);
+      current_scope_initializer_set.insert(it.first);
+    }
+
+
+    // handle outer scope value which is a constant initializer
+    if (include_outer_scope_args) {
+      for (auto& node_idx : graph_view.GetNodesInTopologicalOrder()) {
+        const auto& node = graph_view.GetNode(node_idx);
+        for (const auto& input : node->InputDefs()) {
+          if (current_scope_initializer_set.find(input->Name()) != current_scope_initializer_set.end()) {
+            continue;
+          }
+          if (graph_view.IsConstantInitializer(input->Name(), true)) {
+            auto* p_initializer = graph_proto.add_initializer();
```

Comment:
What happens if the outer scope value is not a constant initializer? Should we output a warning or fail? It's very unlikely, but silently not adding the initializer would be harder to debug if it did happen.